fork_choice/confirmation-rule.md

```diff
@@ -0,0 +1,431 @@
+| Input Parameter                    | Type     | Max. Value                         | Description                                                            |
+| ---------------------------------- | -------- |:---------------------------------- | ---------------------------------------------------------------------- |
+| `confirmation_byzantine_threshold` | `uint64` | `33`                               | the maximum percentage of Byzantine validators among the validator set |
+| `confirmation_slashing_threshold`  | `uint64` | `confirmation_byzantine_threshold` | the maximum percentage of slashings among the validator set            |
+
+
+### Helper Functions
+
+#### `get_committee_weight_between_slots`
+
+```python
+def get_committee_weight_between_slots(store: Store, start_slot: Slot, end_slot: Slot) -> Gwei:
+    """Returns the total weight of committees between ``start_slot`` and ``end_slot`` (inclusive of both).
+    Uses the justified state to compute committee weights.
+    """
+    justified_state = store.checkpoint_states[store.justified_checkpoint]
+    total_active_balance = get_total_active_balance(justified_state)
+
+    # If an entire epoch is covered by the range, return the total active balance
+    start_epoch = compute_epoch_at_slot(start_slot)
+    end_epoch = compute_epoch_at_slot(end_slot)
+    if end_epoch > start_epoch + 1:
+        return total_active_balance
+
+    committee_weight = total_active_balance // SLOTS_PER_EPOCH
+    if start_epoch == end_epoch:
+        num_committees = end_slot - start_slot + 1
+    else:
+        # A range that spans an epoch boundary, but does not span any full epoch
+        # needs pro-rata calculation
+        # First, calculate the number of committees in the current epoch
+        num_slots_in_current_epoch = (end_slot % SLOTS_PER_EPOCH) + 1
+        # Next, calculate the number of slots remaining in the current epoch
+        remaining_slots_in_current_epoch = SLOTS_PER_EPOCH - num_slots_in_current_epoch
+        # Then, calculate the number of slots in the previous epoch
+        num_slots_in_previous_epoch = SLOTS_PER_EPOCH - (start_slot % SLOTS_PER_EPOCH)
+
+        # Each committee from the previous epoch only contributes a pro-rated weight
+        multiplier = remaining_slots_in_current_epoch / SLOTS_PER_EPOCH
+        num_committees = num_slots_in_current_epoch + num_slots_in_previous_epoch * multiplier
+    return num_committees * committee_weight
+```
+
+#### `is_one_confirmed`
+
+```python
+def is_one_confirmed(store: Store, confirmation_byzantine_threshold: int, block_root: Root) -> bool:
+    current_slot = get_current_slot(store)
+    block = store.blocks[block_root]
+    parent_block = store.blocks[block.parent_root]
+    support = int(get_weight(store, block_root))
+    maximum_support = int(get_committee_weight_between_slots(store, Slot(parent_block.slot + 1), current_slot))
+    proposer_score = int(get_proposer_score(store))
```

Comment:
Should `proposer_score` also be scaled with the `multiplier` inside `maximum_support`?
as in, should `proposer_score` be `PROPOSER_SCORE_BOOST * maximum_support div 100` here, instead of always being based on the 1.0 multiplier case?

Reply:
Not according to the algorithm in the paper

Reply:
Hmm, would be nice to double check with the authors, just in case it's an oversight.

Reply:
I confirm that it is correct not to scale the proposer score by the multiplier.